synapse/handlers/sync.py

```diff
@@ -288,6 +288,7 @@ async def wait_for_sync_for_user(
         since_token: Optional[StreamToken] = None,
```

Comment:
Some discussion in `#synapse-dev` about this PR: https://matrix.to/#/!vcyiEtMVHIhWXcJAfl:sw1v.org/$oaV7wGmCQOD-pByNgH6x1YhrVrqHBnEuKD76QQ7bClU?via=matrix.org&via=element.io&via=beeper.com